pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -608,6 +610,29 @@ func (cb *ClusterBuilder) setUpstreamProtocol(cluster *clusterWrapper, port *mod
 		return
 	}
 
+	// Preserve HTTP/1.x traffic header case
+	if port.Protocol.IsHTTP() && cb.req.Push.ProxyConfigs.EffectiveProxyConfig(cb.proxyMetadata, cb.req.Push.Mesh).GetProxyHeaders().GetPreserveHttp1HeaderCase().GetValue() {
```

Comment:
should use metadata.ProxyConfigOrDefault not this method I believe

Reply:
Yep. the EffectiveProxyConfig is about merging the CRD with various other things. ProxyConfigOrDefault properly uses the proxy config of a specific pod, but if it has none set (should ~never happen?) it uses defaults

Reply:
Fixed